docs/modules/ROOT/pages/references/projectsyn/developer.adoc

```diff
@@ -0,0 +1,125 @@
+[NOTE]
+====
+When using the `commodore component new` command to bootstrap a new component, the `component-` prefix will be added automatically and doesn't need to be specified:
+
+[source,console]
+----
+commodore component new openshift4-registry --name "OpenShift 4 Registry" --owner appuio
+----
+====
+
+
+== https://docs.openshift.com/container-platform/latest/operators/understanding_olm/olm-understanding-olm.html[Operator Lifecycle Manager (OLM)]
+
+We try to avoid using the OLM to install operators on a cluster.
+This is mainly due to the added complexity the OLM introduces and us not using any of it's features. This might change in the future.
+
+Exceptions to this rule include cases where the Red Hat documentation explicitly requires the OLM.
+
+Any operators we **do** install with the OLM should be restricted to single namespaces and must never be installed using the `AllNamespaces` InstallMode.
+
+We don't provide any support for operators installed by customers via the OLM.
+
+
+== CustomResource (CR) Specs
+
+Components which configure one or multiple CRs should do so by exposing its `.spec` field as a parameter.
+This enables the configuration of all possible fields via the inventory and provides the maximum flexibility.
+Default values can either be set in the `defaults.yml` file of the component or in JSONNET code.
+It should always be possible to override default values in the inventory.
+
+If creating multiple CRs, a dict should be used where its keys map to the names used for the CRs.
+Iterating over the keys of this dict in JSONNET enables the enrichment of possible default values if not set.
+
+Due to the way http://reclass.pantsfullofunix.net/index.html[reclass] handels `null` overrides, the component must handle `null` values in dicts gracefully.
+
+.main.jsonnet
+[source,jsonnet]
+----
+local ingressControllers =
+  if params.ingressControllers != null
+  then std.objectFields(params.ingressControllers)
+  else [];
+
+{
+  [name]:
+    [kube._Object('operator.openshift.io/v1', 'IngressController', name) {
+      metadata+: {
+        namespace: params.namespace,
+      },
+      spec: {
+        defaultCertificate: {
+          name: acmeCertName,
+        },
+      } + params.ingressControllers[name],
+    }]
+  for name in ingressControllers
+} + {
+  [if std.length(ingressControllers) == 0 then '.gitkeep']: {},
+}
+----
+
+
+== Inventory Parameters
+
+A component can define various parameters in the inventory (`defaults.yml`).
+These should be namespaced in a dict with the component's name (hyphens replaced with underscores):
+
+.defaults.yml
+[source,yaml]
+----
+parameters:
+  openshift4_ingress:
+    namespace: openshift-ingress
+    cloud:
+      provider: ${cloud:provider}
+    ingressControllers:
+      default:
+        replicas: 2
+
+----
+
+If the component requires additional information like the cloud provider, cluster version or infrastructure ID, these can also be accessed via parameters of the inventory.
```

Comment:
Make it clear that this is about facts—or the lack thereof. For me this currently reads confusing.